test/integration/client-side-encryption/client_side_encryption.prose.test.js

```diff
@@ -1723,6 +1723,148 @@ describe('Client Side Encryption Prose Tests', metadata, function () {
+        kmsProviders: getKmsProviders()
+      });
+
+      // Using client_encryption, create a data key with a local KMS provider and the keyAltName "def".
+      setupKeyId = await clientEncryption.createDataKey('local', {
+        keyAltNames: ['def']
+      });
+    });
+
+    afterEach(async () => {
+      clientEncryption = null;
+      setupKeyId = null;
+      await client.close();
+    });
+
+    context('Case 1', metadata, function () {
+      it('createDataKey() handles duplicate key errors on the keyvault collection', async function () {
+        // 1. Use client_encryption to create a new local data key with a keyAltName "abc" and assert the operation does not fail.
+        try {
+          await clientEncryption.createDataKey('local', {
+            keyAltNames: ['abc']
+          });
+        } catch (err) {
+          expect.fail(
+            'Error in step 1) expected clientEncryption.createDataKey to not fail, but received',
+            err
+          );
+        }
+
+        // 2. Repeat Step 1 and assert the operation fails due to a duplicate key server error (error code 11000).
+        try {
+          await clientEncryption.createDataKey('local', {
+            keyAltNames: ['abc']
+          });
+          expect.fail(
+            'Error in step 2) expected clientEncryption.createDataKey to throw duplicate key error but it did not'
+          );
+        } catch (err) {
+          expect(err).have.property('code', 11000);
+        }
+
+        // 3. Use client_encryption to create a new local data key with a keyAltName "def" and assert the operation fails due to a duplicate key server error (error code 11000).
+        try {
+          await clientEncryption.createDataKey('local', {
+            keyAltNames: ['def']
+          });
+          expect.fail(
+            'Error in step 3) expected clientEncryption.createDataKey to throw duplicate key error but it did not'
+          );
+        } catch (err) {
+          expect(err).to.have.property('code', 11000);
+        }
+      });
+    });
+
+    context('Case 2', metadata, function () {
+      it('addKeyAltName() handles duplicate key errors on the keyvault collection', async function () {
+        let _id;
+        // 1. Use client_encryption to create a new local data key and assert the operation does not fail.
+        try {
+          _id = await clientEncryption.createDataKey('local');
+        } catch (err) {
+          expect.fail(
+            'Error in step 1) expected clientEncryption.createDataKey to not fail, but received',
+            err
+          );
+        }
+
+        // 2. Use client_encryption to add a keyAltName "abc" to the key created in Step 1 and assert the operation does not fail.
+        try {
+          await clientEncryption.addKeyAltName(_id, 'abc');
+        } catch (err) {
+          expect.fail(
+            'Error in step 2) expected clientEncryption.createDataKey not to fail, but received',
```

Comment:
this needs to be `addAltKeyName` not `createDataKey` in this and all the custom error messages below